### PR TITLE
CodeCoverage Clover XML

### PR DESCRIPTION
**Pull request by milo**
This is still WIP. I would appreciate if somebody can test it with tools like Jenkins CI.

To try it, call Tester as usually, but set the coverage file extension to xml like:
```
tester --coverage coverage.xml
```

### CODE REVIEW

**Comment by hrach** [comment]
Awesome! :+1:

**Comment by fprochazka** [comment]
:+1:

File: Tester/CodeCoverage/template-clover.phtml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<<?php?>?xml version="1.0" encoding="UTF-8"?>
+<coverage generated="<?php echo time(); ?>">
+	<project>
+		<?php foreach ($files as $id => $info): ?>
+		<file name="<?php echo htmlspecialchars($info->file); ?>">
+			<?php foreach ($info->lines as $line => $arg): if($arg < -1) continue; ?>
```
Comment (JanTvrdik) [inline Tester/CodeCoverage/template-clover.phtml]: missing space after `if`

File: Tester/CodeCoverage/template-clover.phtml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+			<line num="<?php echo intval($line); ?>" count="<?php echo $arg >= 1 ? 1 : 0 ?>"/>
```
Comment (JanTvrdik) [inline Tester/CodeCoverage/template-clover.phtml]: I don't now what may `$arg` contain but isn't it possible to use `min($arg, 1)`?

Comment (matej21) [inline Tester/CodeCoverage/template-clover.phtml]: It [may](http://xdebug.org/docs/code_coverage) contain 1,-1,-2. So `max($arg, 0)` should work.

**Comment by Vrtak-CZ** [comment]
:clap: :flushed:

**Comment by Vrtak-CZ** [comment]
I'm tested this 2 weeks ago but it's not working with Jenkins and Coverals.io so I'm working on it (https://github.com/Vrtak-CZ/tester/compare/clover). Probably be ready before next Testing PoSobota :wink:

**Comment by o5** [comment]
> but it's not working with Jenkins and Coverals.io

Same with Scrutinizer :/ I'll try your fork and maybe I'll save @TomasVotruba 's [work](https://github.com/o5/grido/pull/180#issuecomment-73849473) :)

**Comment by milo** [comment]
@Vrtak-CZ @o5 I have one more branch locally with a rich Clover output. I'll clean in and publish today or tomorrow.

**Comment by milo** [comment]
Sorry guys. Promised for today but must postpone to tomorrow evening :(

**Comment by Vrtak-CZ** [comment]
this should be closed

**Comment by milo** [comment]
Closed in favour of #211

**Pull request closed without merging**